### ISSUE TITLE
(css) Wrong highlight :first-of-type :first-child

### ISSUE THREAD

**Issue by zweroboy1** [highlightjs/highlight.js#3815]
**Describe the issue**
Wrong highlight :first-of-type :first-child, "-of-type" "-child" is not highlighted

**Which language seems to have the issue?**
CSS

**Are you using `highlight` or `highlightAuto`?**
highlightElement

**Sample Code to Reproduce** 
chatGPT
![hljs](https://github.com/highlightjs/highlight.js/assets/65929550/8ade4d79-e299-489d-8c8e-fd4664f28384)

**Expected behavior**
Correct highlighting :)

**Additional context**

**Comment by joshgoebel** [comment]
Please provide textual content of problematic example, not just an image.

**Comment by zweroboy1** [comment]
For the CSS language, each letter in the pseudo-classes ":first-of-class" and ":first-child" should be colored. Currently, only ":first" is colored, while "-of-class" and "-child" remain the default color.

**Comment by joshgoebel** [comment]
Yes, I understand, but please provide the RAW text (that you took a snap of) that I can easily copy and paste for testing.

**Comment by neix604** [comment]
There is the same problem here. **"-of-type"** and **"-child"** are not highlighted.

![css_issue](https://github.com/highlightjs/highlight.js/assets/116621441/69a121bf-76cd-47b7-9109-730b2122e9a9)

The code is as follows.

:first-child {
	padding: 0;
}
:first-of-type {
	padding: 0;
}

**Comment by joshgoebel** [comment]
For use in "either" style regex the word lists need to be reverse-alpha sorted - which has the effect of putting the LONGEST terms first, preventing the shorter words from stealing the match.  So simply `reverse()`ing the keyword lists  might solve this.  I think there are other examples.

**Comment by TimChinye** [comment]
This has not been fixed.

See:
![image](https://github.com/highlightjs/highlight.js/assets/150863066/a843eec2-b96e-4c74-894a-44d3b292fb60)
`span:first-child {`

Additionally, it doesn't recognise:
![image](https://github.com/highlightjs/highlight.js/assets/150863066/671a2e48-42bd-492d-b3bd-74a8d128e198)
`span::first-child {`

**Comment by Dxuian** [comment]
is this still open ?

**Comment by joshgoebel** [comment]
yes appears so

**Comment by joshgoebel** [comment]
Fixed by sorting pseudo-class/element keyword lists longest-first (`.sort().reverse()`) so shorter names like `first` no longer steal the match from `first-child` / `first-of-type`.

Verified on current `main` — e.g. `:first-child` and `:first-of-type` are fully highlighted as `selector-pseudo`.
